version: '3.8'
services:
  user_database:
    image: postgres:16.1-alpine3.19
    container_name: user_database
    restart: always
    env_file:
      - .env
    volumes:
      - data:/var/lib/postgresql/data
      - ./user-db/src/init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - '5432:5432'
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "${POSTGRES_USER}"]
      interval: 10s
      timeout: 5s
      retries: 5

  pg_admin:
    image: dpage/pgadmin4:7.4
    container_name: pg_admin
    restart: always
    env_file:
      - .env
    ports:
      - '${PGADMIN_LISTEN_PORT}:5050'
    volumes:
      - pgadmin:/var/lib/pgadmin
    depends_on:
      - user_database

  reviews_minio:
    image: quay.io/minio/minio:RELEASE.2024-02-09T21-25-16Z
    command: server --console-address ":9001" /data_reviews
    ports:
      - "9000:9000"
      - "9001:9001"
    hostname: reviews_minio
    volumes:
      - data_reviews1:/data1
      - data_reviews2:/data2
    healthcheck:
      test: ["CMD", "mc", "ready", "local"]
      interval: 5s
      timeout: 5s
      retries: 5

volumes:
  data:
    driver: local
  data_reviews1:
    driver: local
  data_reviews2:
    driver: local
  pgadmin:
    driver: local
